utils/prof_utils.py

```diff
@@ -0,0 +1,1177 @@
+        plt.xticks(fontsize=18)
+        plt.yticks(fontsize=18)
+        plt.xlim(0, len(std_prof))
+        plt.xlabel("Bins", fontsize=20)
+        plt.ylabel("Intensity", fontsize=20)
+        for i, mx in enumerate(maxima):
+            plt.axvline(x=(mx + maxima_e[i])/len(profile), ls=":", lw=2, color="gray")
+            plt.axvline(x=(mx - maxima_e[i])/len(profile), ls=":", lw=2, color="gray")
+        plt.plot(std_prof, label="Original Profile", color="black")
+        plt.plot(on_pulse_prof, label="Regularized profile", color="red")
+        plt.legend(loc="upper right", prop={'size': 16})
+        plt.savefig(plot_name)
+
+    fit_dict = {"W10":W10, "W10_e":W10_e, "W50":W50, "W50_e":W50_e, "Wscat":Wscat, "Wscat_e":Wscat_e,
+                "Weq":Weq, "Weq_e":Weq_e, "maxima":maxima, "maxima_e":maxima_e}
+    return fit_dict
+
+#---------------------------------------------------------------
+def prof_eval_gfit(profile, max_N=6, ignore_threshold=0.02, plot_name=None, min_comp_len=None, alpha=2):
+    """
+    Fits multiple gaussians to a profile and subsequently finds W10, W50, Weq and maxima
+
+    Parameters:
+    -----------
+    profile: list
+        The pulse profile to evaluate
+    chi_threshold: float
+        OPTIONAL - The script will stop trying new fits when the reduced chi-squared is within this amount to unity. Default: 0.05
+    plot_name: string
+        OPTIONAL - If not none, will make a plot of the best fit with this name. Default: None
+    ignore_threshold: float
+        OPTIONAL -  Maxima with values below this number will be ignored. Default: 0.02
+    min_comp_len: float
+        OPTIONAL - Minimum length of a component to be considered real. Measured in bins. If none, will use 1% of total profile lengths. Default: None
+
+    Returns:
+    --------
+    [W10, W10_e, W50, W50_e, Weq, Weq_e, maxima, maxima_e]: list
+        W10: float
+            The W10 width of the profile measured in number of bins
+        W10_e: float
+            The uncertainty in the W10
+        W50: float
+            The W50 width of the profile measured in number of bins
+        W50_e: float
+            The uncertainty in the W50
+        Weq: float
+            The equivalent width of the profile measured in number of bins
+        Weq_e: float
+            The uncertainty in the equivalent width
+        maxima_e: list
+            A list of floats, each correspinding to the error of the maxima of the same index. Measured in bins
+    """
+    #initialize minimum component length
+    if min_comp_len is None:
+        min_comp_len = int(len(profile)/100)
+
+    #Normalize, find the std
+    y = np.array(profile)/max(profile)
+    noise_std, clipped = sigmaClip(y, alpha=alpha)
+    check_clip(clipped, toomuch=0.8, toolittle=0.)
+
+    y = y - np.nanmean(clipped)
+    y = y/max(y)
+
+    #fit gaussians
+    fit, chisq, bic, popt, pcov = fit_gaussian(y, max_N=max_N, min_comp_len=min_comp_len, alpha=alpha)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused variable 'bic'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4645369)